### ISSUE TITLE
Release Builds: Develop an Automated Release Build

### ISSUE THREAD

**Issue by JackEAllen** [ISISComputingGroup/IBEX#6973]
As a developer, preparing for a release is quite a time consuming task which could be automated to allow developers to focus efforts on other tasks.

I would like to write a script which runs as a jenkins job that automates the [pre-testing steps](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Creating-a-release#pre-testing) part of creating a release. It'll need to create branches, write version numbers and tag with the relevant version. After this is done it should trigger the relevant release builds. 

This will probably need a jenkins arg for the version number to create. 

### Acceptance Criteria
_What is the acceptance criteria?_
- [ ] A jenkins job is created for automating the pre-testing steps of a release which can be. This is tested with a release called x.x.x or something so we don't actually deploy it accidentally. 

### Extra Information
_Where required files/links are_
See release instructions for more information about releases: https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Creating-a-release

## How to Test
_verbose instructions for reviewer to test changes
(**Add before making a PR**)_

**Comment by rerpha** [comment]
Could we have a Jenkins job that will checkout the code, create branches and push them, then we could just do the usual "scan repository now" or even just get the job to trigger them. We could also maybe do with another (post-testing) job that will delete the release branches and tag them

**Comment by rerpha** [comment]
As mentioned by Freddie: like the repo checks this would be best suited for one of the Linux nodes as they're a lot faster than some windows for git operations

**Comment by FreddieAkeroyd** [comment]
The instructions also say to use `--remote` to `sets all the repos to their pinned version`, but this actually does the opposite - it updates to tip of branch

**Comment by FreddieAkeroyd** [comment]
Hi @NikolaRoev thank you for your work in this, the `Release_branches` job runs through and pushes all the correct branches and did all this in build 23 with `REMOTE==false`. If I set `REMOTE=true` (as in build 22) it fails as some of the submodules also have submodules and the `--recursive` flags means these error if  out of date. I am not sure yet of the best approach - we may have pinned our version specifically, or we may have forgotten to update the other submodule. We can handle the former by not using `--recursive` so it just checks our submodule is up to date but not any dependencies, but i need to investigate the moduels concerned as to what we really want. So i am happy to leave this as is for the moment as the module is basically working. 

The only issue I have found is that some of the edits don't seem to quite work. I created a test 22.22.22 release and:
- https://github.com/ISISComputingGroup/ibex_gui/tree/Release_22.22.22 has an incorrect version format in the manifest
- https://github.com/ISISComputingGroup/EPICS/tree/Release_22.22.22 the version in   svn-revision.db.tmpl is not correct
